src/probnum/filtsmooth/gaussfiltsmooth/extendedkalman.py

```diff
@@ -1,102 +1,151 @@
 """Gaussian filtering and smoothing based on making intractable quantities tractable
 through Taylor-method approximations, e.g. linearization."""
 
+import typing
+
 import numpy as np
 
 import probnum.random_variables as pnrv
+import probnum.type as pntype
 from probnum.filtsmooth import statespace
 
+from .linearizing_transition import LinearizingTransition
 
-class ContinuousEKFComponent(statespace.Transition):
-    """Continuous extended Kalman filter transition."""
-
-    def __init__(self, non_linear_sde, num_steps):
-        if not isinstance(non_linear_sde, statespace.SDE):
-            raise TypeError("Continuous EKF transition requires a (non-linear) SDE.")
-        self.non_linear_sde = non_linear_sde
-        self.num_steps = num_steps
-        super().__init__()
 
-    def transition_realization(
-        self, real, start, stop, _linearise_at=None, _diffusion=1.0, **kwargs
-    ):
+class EKFComponent(LinearizingTransition):
+    """Interface for extended Kalman filtering components."""
 
-        compute_jacobian_at = _linearise_at.mean if _linearise_at is not None else real
+    def __init__(self, non_linear_model) -> None:
+        super().__init__(non_linear_model=non_linear_model)
 
-        def jacobfun(t, x=compute_jacobian_at):
-            # replaces functools (second variable may not be called x)
-            return self.non_linear_sde.jacobfun(t, x)
+        # Will be constructed later
+        self.linearized_model = None
 
-        step = (stop - start) / self.num_steps
-        return statespace.linear_sde_statistics(
-            rv=pnrv.Normal(mean=real, cov=np.zeros((len(real), len(real)))),
-            start=start,
-            stop=stop,
+    def transition_realization(
+        self,
+        real: np.ndarray,
+        start: float,
+        stop: typing.Optional[float] = None,
+        step: typing.Optional[float] = None,
+        _linearise_at: typing.Optional[pnrv.RandomVariable] = None,
+        _diffusion: typing.Optional[pntype.FloatArgType] = 1.0,
+        **kwargs
+    ) -> (pnrv.Normal, typing.Dict):
+
+        # Do we want the EKF to linearise first and then transition the real,
+        # or try to call the non-linear transition if possible?
+        # There is no right or wrong, but what is the desired behaviour?
```

Comment:
The way I see it, the EKF is only really needed when we transition a RV. For an actual realization, we don't need to approximate anything. Does that make sense? Or did I maybe completely miss the point? :D

Reply:
No you did not miss a point as far as I know :) This is precisely my concern.
I think the cases we need to keep in mind here is what happens if (a) the non-linear transition allows transition_realization (in which case we want to call this, I suppose?) and (b) if it does not allow it (do we want an error, or do we want to call the extended kalman?. Currently, the code uses case (a); I am not sure 100% what happens in case (b). We can also leave this in here, and solve it in the future. I dont know what the best solution would be.